compiler/dyno/tools/chpldoc/chpldoc.cpp

```diff
@@ -2119,16 +2084,34 @@ void generateSphinxOutput(std::string sphinxDir, std::string outputDir,
 
 
 int main(int argc, char** argv) {
-
-  init_args(&sArgState, argv[0]);
-  init_arg_desc(&sArgState, docs_arg_desc);
-  process_args(&sArgState, argc, argv);
+  // initial value of CHPL_HOME may be overridden by cmdline arg
+  CHPL_HOME = getenv("CHPL_HOME");
   Args args = parseArgs(argc, argv);
-  // add source files
-  // TODO: Check for proper file type, duplicate file names, was file found, etc.
-  for (int i = 0; i < sArgState.nfile_arguments; i++) {
-    args.files.push_back(std::string(sArgState.file_argument[i]));
+
+  // check if user asked for legacy chpldoc
+  if (fLegacyChpldoc) {
+    std::string pathToExe = getExecutablePath(argv[0], nullptr);
```

Comment:
pass `&main` as the second argument

Reply:
passed, although `&main` complained a lot about incompatible types, `(void*)main` worked